main.tf

```diff
@@ -63,7 +63,6 @@ resource "aws_s3_bucket" "default" {
   provider      = aws.virginia
   bucket        = var.domain_name
   force_destroy = var.force_destroy
-  policy        = var.policy
```

Comment:
Have you considered adding a `aws_s3_bucket_policy` resource to avoid dropping the bucket policy altogether?

Reply:
We are not dropping it, you can easily enable it by creating `aws_s3_bucket_policy` outside this module and using `bucket_id` to link policy with it.  There is no point IMHO to pass JSON as a variable just to create this.  Also, someone might want to create two policies or something.  It is a breaking change, though, but not a big one, and everyone who would like to consider switching to v4 aws provider will need to put work anyway.

Does it make sense?